src/dune_sexp/decoder.mli

```diff
@@ -189,6 +189,10 @@ val plain_string : (loc:Loc.t -> string -> 'a) -> 'a t
 (** A valid filename, i.e. a string other than "." or ".." *)
 val filename : Filename.t t
 
+(** An extension: a string not starting with ".". The value returned by the
+    parser is prefixed with ".". *)
+val extension : string t
```

Comment:
I think this can be `Filename.Extension.t t`

Reply:
I am not sure about this one because `Filename.Extension.t` is related to the `Filename.extension` function which is a different notion (it does not admit multiple period-separated components).